Source/WebGPU/WGSL/EntryPointRewriter.cpp

```diff
@@ -141,12 +138,10 @@ void EntryPointRewriter::constructInputStruct()
     // insert `var ${parameter.name()} = ${structName}.${parameter.name()}`
     AST::StructMember::List structMembers;
     for (auto& parameter : m_parameters) {
-        auto parameterType = adoptRef(*new AST::TypeReference(m_emptySourceSpan, parameter.m_type.copyRef()));
-
         structMembers.append(makeUniqueRef<AST::StructMember>(
             m_emptySourceSpan,
-            parameter.m_name,
-            parameterType.copyRef(),
+            WTFMove(parameter.m_name),
```

Comment:
Moving from the parameter sounds scary, are you sure you want the parameter's name to be empty after this?

Maybe you want String(parameter.m_name)? Or just change the constructor to StructMember to accept a String& in addition to a String&&?

Reply:
The variable naming is a bit confusing, but `parameter` here refers to a struct that I build as part of this pass only for this purpose, so I'm sure we won't need it after.